src/artifacts.js

```diff
@@ -509,6 +509,16 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
       key: artifact.details.key,
     };
 
+    res.set('x-amz-meta-runid', runId);
+    res.set('x-amz-meta-taskid', taskId);
+    res.set('x-amz-meta-name', name);
+    res.set('x-amz-meta-content-sha256', artifact.details.contentSha256);
```

Comment:
I can't see anywhere this header is set?

Reply:
https://github.com/taskcluster/taskcluster-queue/blob/f53a106b7768a8bcd9b01e7664ad3a00575bc8a9/src/artifacts.js#L204

Reply:
that sets `x-amz-content-sha256`

Reply:
nope, x-amz-content-sha256 is the sha256 of the request for the S3 request but is not persisted as an object header.  It is actually closer in relation to x-amz-meta-transfer-sha256 than x-amz-meta-content-sha256.  In single part uploads, it does match exactly x-amz-meta-transfer-sha256 but in multipart it matches per request the sha256 of the content-encoded parts.